### PR TITLE
boards: nrf54l15pdk_nrf54l15_cpuapp: add support for PDK 0.7.0

### PR DESCRIPTION
**Pull request by barsok**
Add explicite support for the PDKs in all revisions

### CODE REVIEW

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuflpr_0_7_0.overlay [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nrf54l15pdk_nrf54l15_common_0_3_0.dtsi"
```
Comment (masz-nordic) [inline boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuflpr_0_7_0.overlay:7]: Same with 0.3.0 vs 0.7.0

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_7_0.overlay [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nrf54l15pdk_nrf54l15_common_0_3_0.dtsi"
```
[the remark this answers is not in the document]
Comment (magp-nordic) [reply boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_7_0.overlay:7]: @barsok if 0.7.0 has the same pinout for LEDs and buttons as 0.3.0, then file `nrf54l15pdk_nrf54l15_common_0_3_0.dtsi` should most probably be renamed to something that will indicate that it is common for 0.3.0 and 0.7.0, for example, `nrf54l15pdk_nrf54l15_common_0_3_0_S_0_7_0.dtsi` where 'S' stands for '/' (it does not look good, but it is the best one I came up with for now). Otherwise, the fact that 0.7.0 overlay includes dtsi for 0.3.0 may be confusing for some people.

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml
Comment (magp-nordic) [inline boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml:1]: This file should probably be added in a separate commit. It is not needed for 0.7.0 support, but to support 0.2.1 in tests.

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+identifier: nrf54l15pdk@0.2.1/nrf54l15/cpuapp
+name: nRF54l15-PDK-nRF54l15-Application
+type: mcu
+arch: arm
+toolchain:
+  - gnuarmemb
+  - xtools
+  - zephyr
+ram: 256
+flash: 1536
```
Comment (magp-nordic) [inline boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml:13]: Are those values correct? I think those values should be equal to the ones specified in `nrf54l15pdk_nrf54l15_cpuapp.yaml`, as that is how much RAM and NVM is assigned to APP core.

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_7_0.overlay (hunk 2 above)
[the remark this answers is not in the document]
Comment (barsok) [reply boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_7_0.overlay:8]: Thank you for this suggestion. I have added comment for 0.7.0, maybe this is acceptable? Checked other vendors and there are cases only for "ns" environment where the same file is included. Here the 0.7.0 acts as a label for 0.3.0.

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml
Comment (barsok) [reply boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml:1]: My main point was to change handling of the boards (no default device tree, move all pin assignments to overlays) and add the 0.7. Please let me know if this is Zephyr standard way so then I will create a PR with rearranging 0.2.1 and 0.3.0 and a separate one with only added 0.7.0.

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml (hunk 3 above)
Comment (barsok) [reply boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml]: Thank you, I have updated the numbers for all the boards

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuflpr_0_7_0.overlay (hunk 1 above)
Comment (kl-cruz) [reply boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuflpr_0_7_0.overlay:8]: Eventually to name it for example config_v1 like: nrf54l15pdk_nrf54l15_common_config_v1.dtsi and include it into 0.3.0 and 0.7.0

**Review by masz-nordic — DISMISSED** [review]

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_7_0.overlay [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* PDK 0.7.0 is functionally equivalent to 0.3.0 */
+#include "nrf54l15pdk_nrf54l15_common_0_3_0.dtsi"
```
Comment (anangl) [inline boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_7_0.overlay:8]: What's the point in adding 0.7.0 then? How does it differ from 0.3.0?

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml
Comment (magp-nordic) [reply boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml:1]: What I meant was that it looks like the yaml file for 0.2.1 was added, because it was missing before, not because it needed to be added since 0.7.0 was added. If that is the case, then yaml file for 0.2.1 should be added in a separate commit, since it is not related to the commit title, which is "add support for 0.7.0 PDK (same goes for fixing size of ram and flash in yaml files). There is no need to create another PR.
Alternatively, the commit message could be changed: the title could be something like "Add explicite support for all PDKs" and then the commit body can have explanation that it adds support for 0.7.0 and fixes yamls for other revisions.
The reason is that in current shape, it might be confusing for someone searching through commit history that according to commit title it does one thing, but it also does something else. I think commit should do only what its written in its title, just like functions should only do what their name is suggesting.
I looked into Zephyr Contribution Guidelines, but I haven't found anything about that, so I guess it is just my nitpicking 😄

**Review by magp-nordic — DISMISSED** [review]
I left one comment, but it is not a blocker, but just a minor improvement.

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_common_0_2_1.dtsi
Comment (anangl) [inline boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_common_0_2_1.dtsi:1]: Since this file kind of overlays `nrf54l15pdk_nrf54l15-common.dtsi`, it would be better to name it `nrf54l15pdk_nrf54l15-common_0_2_1.dtsi` (with a hyphen before "common").

File: boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml [hunk 5]
```diff
@@ -1,16 +1,16 @@
 # Copyright (c) 2024 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
-identifier: nrf54l15pdk@0.3.0/nrf54l15/cpuapp
+identifier: nrf54l15pdk@0.2.1/nrf54l15/cpuapp
```
Comment (anangl) [inline boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuapp_0_2_1.yaml:4]: Aren't such `_0_2_1.yaml` files also needed for `cpuflpr`?

**Review by anangl — DISMISSED** [review]

**Review by magp-nordic — DISMISSED** [review]

**Review by masz-nordic — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by kl-cruz — APPROVED** [review]
